server/controllers/events/gateway_command_executor.go

```diff
@@ -0,0 +1,85 @@
+package events
+
+import (
+	"bytes"
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/events"
+	"github.com/runatlantis/atlantis/server/events/models"
+	"github.com/runatlantis/atlantis/server/logging"
+	"github.com/runatlantis/atlantis/server/lyft/aws/sns"
+	"github.com/uber-go/tally"
+	"net/http"
+	"time"
+)
+
+type GatewayCommandExecutor struct {
+	SNSWriter     sns.Writer
+	CommandRunner events.CommandRunner
+	Scope         tally.Scope
+	Logger        logging.SimpleLogging
+}
+
+func (g *GatewayCommandExecutor) ExecuteCommentCommand(request *http.Request, _ models.Repo, _ *models.Repo, _ *models.PullRequest, _ models.User, _ int, _ *events.CommentCommand, _ time.Time) HttpResponse {
+	err := g.SendToWorker(request)
+	if err != nil {
+		g.Logger.With("err", err).Warn("Failed to send comment request to Atlantis worker")
+		return HttpResponse{
+			body: err.Error(),
+			err: HttpError{
+				code: http.StatusBadRequest,
+				err:  err,
+			},
+		}
+	}
+	return HttpResponse{
+		body: "Processing...",
+	}
+}
+
+func (g *GatewayCommandExecutor) ExecuteAutoplanCommand(request *http.Request, eventType models.PullRequestEventType, baseRepo models.Repo, headRepo models.Repo, pull models.PullRequest, user models.User, _ time.Time, _ logging.SimpleLogging) HttpResponse {
+	var err error
+	var respBody string
+	switch eventType {
+	case models.OpenedPullEvent, models.UpdatedPullEvent:
+		// If the pull request was opened or updated, we perform a pseudo-autoplan to determine if tf changes exist.
+		// If they exist, then we will forward request to the worker.
+		if hasTerraformChanges := g.CommandRunner.RunPseudoAutoplanCommand(baseRepo, headRepo, pull, user); hasTerraformChanges {
```

Comment:
I don't think this worth running here. We should just put requests into the worker queue and move on. CommandRunner should figure out the rest.

Reply:
Ignore the comment, this was part of the requirements in the tech spec.